src/main/kotlin/three/consulting/epoc/service/ProjectService.kt

```diff
@@ -0,0 +1,67 @@
+package three.consulting.epoc.service
+
+import mu.KotlinLogging
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.dao.EmptyResultDataAccessException
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import three.consulting.epoc.dto.ProjectDTO
+import three.consulting.epoc.entity.Project
+import three.consulting.epoc.repository.ProjectRepository
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class ProjectService(private val projectRepository: ProjectRepository) {
+
+    fun findProjectForId(id: Long): ProjectDTO? {
+        logger.info { "Looking for project with id: $id" }
+        val project: Project? = projectRepository.findByIdOrNull(id)
+        if (project != null)
+            return ProjectDTO(project)
+        logger.info { "No project found for id: $id" }
+        return null
+    }
+    fun createProject(projectRequest: ProjectDTO): ProjectDTO {
+        logger.info { "Creating new project" }
+        if (projectRequest.id == null) {
+            val project = Project(projectRequest)
+            try {
+                return ProjectDTO(projectRepository.save(project))
+            } catch (e: DataIntegrityViolationException) {
+                throw UnableToCreateProjectException("Cannot create a project with non-existing relation")
```

Comment:
Log here the the DataIntegrityViolationException before throwing the new custom exception